### ISSUE TITLE
Protected memory must not be accessed from vertex stages without protectedNoFault

### ISSUE THREAD

**Issue by olvaffe** [KhronosGroup/Vulkan-ValidationLayers#11532]
**Describe the situation in which you encountered the missing validation**

Skia may use a protected index buffer for drawing. The spec on the other hand says

> If VkPhysicalDeviceProtectedMemoryProperties::protectedNoFault is VK_FALSE, applications must not perform any of the following operations:
> 
> - Write to unprotected memory within protected queue operations.
> - Access protected memory within protected queue operations other than in framebuffer-space pipeline stages, the compute shader stage, in video decode operations, in video encode operations, or the transfer stage.
> - Perform a query within protected queue operations.

Skia violates the second bullet point, but the issue is not caught by VVL.

**Valid Usage IDs requested**

The rules are in the first paragraph of

https://docs.vulkan.org/spec/latest/chapters/memory.html#memory-protected-access-rules

but there is no VUID.

**Additional Context**

**Comment by spencer-lunarg** [comment]
There is a VU like ` VUID-vkCmdDraw-commandBuffer-02707`

The issue this is very broad, there are MANY ways to "access protect memory" and we try cover all I know

Can you give more details on **how** Skia is "accessing"

**Comment by olvaffe** [comment]
In the skia case, the command buffer is protected (` VUID-vkCmdDraw-commandBuffer-02707` does not apply) and it binds a protected buffer as the index buffer for drawing. That triggers access violation on mali.

This is a sample test that simulates skia's behavior: https://github.com/olvaffe/gputest/blob/main/vk/protected.c.

**Comment by spencer-lunarg** [comment]
ok, so this is strange, I just removed "index buffer" checks in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/11395 because we now have the VU as

<img width="827" height="352" alt="Image" src="https://github.com/user-attachments/assets/66da3d22-9108-4fa4-8ead-bdeb6286b26a" />

@Tobski @janharaldfredriksen-arm @djdeath do you know what is going on here, I was told in https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/7934#note_580998 that Index buffers are not even allowed to be protected

**Comment by djdeath** [comment]
I thought they were, but that was before I was pointed to the spec bit that says pre-raster stages should not.

**Comment by Tobski** [comment]
@spencer-lunarg Index buffers were supposed to be excluded before, but they weren't. They are now _correctly_ excluded. You shouldn't be able to create a protected index buffer.

**Comment by spencer-lunarg** [comment]
> You shouldn't be able to create a protected index buffer.

Yes, @olvaffe so I now Skia will now just hit ` VUID-VkBufferCreateInfo-flags-09641` at

```c++
    test->ib = vk_create_buffer_with_mt(
        vk, test->protected ? VK_BUFFER_CREATE_PROTECTED_BIT : 0, ib_size,
        VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_INDEX_BUFFER_BIT,
        test->protected ? protected_mt : vk->buf_mt_index);
```

so I don't think there is any missing validation for access time, there just was (recently in the last 2 weeks fixed) nothing checking at buffer creation time

**Comment by olvaffe** [comment]
Yeah, I was using the latest prebuilt android binary. Let me build from tot to verify before I close this issue. Thanks for the prompt responses!

**Comment by olvaffe** [comment]
Confirmed fixed.
